tests/ui/parser/token-splitting.rs

```diff
@@ -0,0 +1,33 @@
+// Check some places in which we want to split multi-character punctuation.
+//@ edition: 2015
+#![feature(never_type)] // only used inside `bang_eq_never_ty`!
+
+//@ check-pass
+
+fn plus_eq_bound() {
+    // issue: <https://github.com/rust-lang/rust/issues/47856>
+    struct W<T: Clone + = ()> { t: T }
+    struct S<T: Clone += ()> { t: T }
+
+    // Bare & `dyn`-prefixed trait object types take different paths in the parser.
+    // Therefore, test both branches.
+
+    #[cfg(false)] let _: Debug + = *(&() as &dyn Debug);
+    #[cfg(false)] let _: Debug += *(&() as &dyn Debug);
+
+    #[cfg(false)] let _: dyn Debug + = *(&() as &dyn Debug);
+    #[cfg(false)] let _: dyn Debug += *(&() as &dyn Debug);
```

Comment:
These are actually semantically well-formed, too. Remove the `#[cfg(false)]`s after imports.